src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs

```diff
@@ -144,19 +206,40 @@ public override void ExecuteCmdlet()
             {
                 VirtualMachine vmParameters = (this.ComputeClient.ComputeManagementClient.VirtualMachines.Get(this.ResourceGroupName, this.VMName));
 
-                bool osVolumeEncrypted = IsOsVolumeEncrypted(vmParameters);
+                EncryptionStatus osVolumeEncrypted = IsOsVolumeEncrypted(vmParameters);
                 DiskEncryptionSettings osVolumeEncryptionSettings = GetOsVolumeEncryptionSettings(vmParameters);
-                bool dataVolumesEncrypted = AreDataVolumesEncrypted(vmParameters);
+                EncryptionStatus dataVolumesEncrypted = AreDataVolumesEncrypted(vmParameters);
 
-                AzureDiskEncryptionStatusContext encryptionStatus = new AzureDiskEncryptionStatusContext
+                OSType osType = GetOSType(vmParameters);
+                switch (osType)
                 {
-                    OsVolumeEncrypted = osVolumeEncrypted,
-                    OsVolumeEncryptionSettings = osVolumeEncryptionSettings,
-                    DataVolumesEncrypted = dataVolumesEncrypted
-                };
-                WriteObject(encryptionStatus);
+                    case OSType.Windows:
+                        AzureDiskEncryptionStatusContext encryptionStatus = new AzureDiskEncryptionStatusContext
+                        {
+                            OsVolumeEncrypted = osVolumeEncrypted,
+                            OsVolumeEncryptionSettings = osVolumeEncryptionSettings,
+                            DataVolumesEncrypted = dataVolumesEncrypted
+                        };
+                        WriteObject(encryptionStatus);
+                        break;
+                    case OSType.Linux:
+                        AzureDiskEncryptionStatusLinuxContext encryptionStatusLinux = new AzureDiskEncryptionStatusLinuxContext
+                        {
+                            OsVolumeEncrypted = osVolumeEncrypted,
+                            OsVolumeEncryptionSettings = null,
```

Comment:
This will be available later?  It is awkward that the LinuxContext class has this property and then it is always set to null.